static/js/components/Button.jsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import PropTypes from "prop-types";
+import MuiButton from "@mui/material/Button";
+import LoadingButton from "@mui/lab/LoadingButton";
+
+const Button = ({ primary, secondary, async, ...muiButtonProps }) => {
+  if (muiButtonProps.startIcon) {
+    throw new Error(
+      "Error: startIcon used in Button props. Please use endIcon as specified in the SkyPortal style documentation: https://skyportal.io/docs/styling.html#buttons"
+    );
+  }
+  if (muiButtonProps.variant || muiButtonProps.color) {
+    throw new Error(
+      "Error: do use variant or color prop in Button, see button documentation for recommended usage https://skyportal.io/docs/styling.html#buttons"
+    );
```

Comment:
I wonder why we should not allow the user to choose a color. What if there's a good reason to want a customization on top of the defaults offered here?

Reply:
I guess my thought was to have an opinionated set of colors (i.e. the default buttons), and if there was need for a different type of button with a different color, a new button type could be added. So you can define the styles for a certain button type in one place and not have a bunch of one-off button types.

Reply:
That's a great idea in general. I am not sure if we would want to have people making up green buttons or something like that... on the other hand sometimes you just want a different color for a good reason. I'm not sure what the best practice would be here. 
My hunch is that relying on the programmer's laziness to just use the defaults is ok, and let the programmer that really wants their red button to be able to use it.

Reply:
Agreed, if we notice many one-off button types are similar we can add more to variants to this Button.jsx component.